### ISSUE TITLE
Deploy to openshift and heroku is not working

### ISSUE THREAD

**Issue by tribet84** [angular-fullstack/generator-angular-fullstack#1920]
After triggering the deployment  functionality for both heroku and openshift nothing happens. I have them properly configured in my machine so that should be fine. Is there any extra logging I can add to this ticket?

This is the actual console output after hit enter:
λ yo angular-fullstack:openshift
(!) NamedBase constructor is deprecated. See https://github.com/yeoman/generator/issues/882
? Name to deploy as: guru

λ yo angular-fullstack:heroku
(!) NamedBase constructor is deprecated. See https://github.com/yeoman/generator/issues/882
? Name to deploy as (Leave blank for a random name): guru

Item | Version
----- | -----
generator-angular-fullstack | 3.7.3
Node | 4.4.4
npm | 3.9.0
Operating System |  Windows 10 

Item | Answer
----- | -----
Transpiler | Babel
Markup |  Jade
CSS |  LESS 
Router |  ui-router
Build Tool | / Gulp
Client Tests | Jasmine / Mocha
DB | MongoDB 
Auth | Y

**Comment by savli** [comment]
I had a very similar issue, which I mentioned in Issue #1916 . What I ended up doing is creating my Heroku App via the heroku.com website, then following the directions for CLI (i.e. creating a git remote) in the `dist` folder. Then, if you want to push any changes to your Heroku app, just use `grunt buildcontrol:heroku`

**Comment by dudeletslive** [comment]
@savli - interestingly, this seems to solve the issue for me as well, though it refused to accept my remote. I had to replace 'heroku' in the Gulpfile with the full path to my app on Heroku.

**Comment by tribet84** [comment]
Ok, I got it working now. This is what I did:

1. Create two config variables on Heroku:
 1.1 NODE_ENV (value 'production' no capital)
 1.2 MONGODB_URI (database connection string better than change my code)
2. Add a buildpack:  heroku buildpacks:
_set https://github.com/heroku/heroku-buildpack-nodejs#v90 -a my-app_
3. Go into the dist folder after building it (**gulp build**) and push the folder into heroku (git init / add / commit / push)

Boom!
